factory/glideFactoryMonitoring.py

```diff
@@ -1093,18 +1123,59 @@ def write_file(self, monitoringConfig=None):
                                                     val_dict_counts_desc, self.updated, val_dict_counts)
             monitoringConfig.write_rrd_multi("%s/Log_Completed" % fe_dir,
                                              "ABSOLUTE", self.updated, val_dict_completed)
+            monitoringConfig.write_completed_json("%s/Log_Completed" % fe_dir, self.updated, val_dict_completed)
             monitoringConfig.write_rrd_multi("%s/Log_Completed_Stats" % fe_dir,
                                              "ABSOLUTE", self.updated, val_dict_stats)
+            monitoringConfig.write_completed_json("%s/Log_Completed_Stats" % fe_dir, self.updated, val_dict_stats)
             # Disable Waste RRDs... WasteTime much more useful
             #monitoringConfig.write_rrd_multi("%s/Log_Completed_Waste"%fe_dir,
             #                                 "ABSOLUTE",self.updated,val_dict_waste)
             monitoringConfig.write_rrd_multi("%s/Log_Completed_WasteTime" % fe_dir,
                                              "ABSOLUTE", self.updated, val_dict_wastetime)
+            monitoringConfig.write_completed_json("%s/Log_Completed_WasteTime" % fe_dir, self.updated, val_dict_wastetime)
+
 
+        self.aggregate_frontend_data(self.updated, diff_summary)
 
         self.files_updated = self.updated
         return
 
+    def aggregate_frontend_data(self, updated, diff_summary):
+        """
+        This goes into each frontend in the current entry and aggregates
+        the completed/stats/wastetime data into completed_data.json
+        at the entry level
+        """
+
+        entry_data = {'frontends':{}}
+
+        for frontend in diff_summary.keys():
+            fe_dir = "frontend_" + frontend
+
+            completed_filename=os.path.join(monitoringConfig.monitor_dir, fe_dir) + "/Log_Completed.json"
+            completed_stats_filename=os.path.join(monitoringConfig.monitor_dir, fe_dir) + "/Log_Completed_Stats.json"
+            completed_wastetime_filename=os.path.join(monitoringConfig.monitor_dir, fe_dir) + "/Log_Completed_WasteTime.json"
+
+            try:
+
+                completed_fp = open(completed_filename)
+                completed_stats_fp = open(completed_stats_filename)
+                completed_wastetime_fp = open(completed_wastetime_filename)
```

Comment:
Again, please make sure to use the correct ways to handle files

Reply:
Done